File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/ECGetRequest.cs
#region "Imports"

using System;
using PayPal.Payments.Common ;
using PayPal.Payments.Common.Utility ;
#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// Used for ExpressCheckout GET operation.
	/// </summary>
	/// <remarks>
	/// <seealso cref="ECSetRequest"/>
	/// <seealso cref="ECDoRequest"/>
	/// </remarks>
	public class ECGetRequest : ExpressCheckoutRequest
	{
		#region "Constructor"
		/// <summary>
		/// Constructor for ECGetRequest
		/// </summary>
		/// <param name="Token">String</param>
		/// <remarks>
		/// ECGetRequest is used to set the data required for a Express Checkout GET operation.
		/// </remarks>
		/// <example>
		/// <code lang="C#" escaped="false">
		///		.............
		///		
		///		//Create the ECGetRequest object
		///		ECGetRequest GetEC = new ECGetRequest("[tokenid]");
		///		
		///		.............
		/// </code>
		/// <code lang="Visual Basic" escaped="false">
		///		.............
		///		
		///		//Create the ECGetRequest object
		///		Dim GetEC As ECGetRequest = new ECGetRequest("[tokenid]")
		///		
		///		.............
		/// </code>
		/// </example>
		public ECGetRequest(String Token) : base(PayflowConstants.ACTIONTYPE_GET, Token)
		{			
		}
		protected ECGetRequest(String Token, String Action) : base(PayflowConstants.ACTIONTYPE_GETBA, Token)
		{			
		}
		#endregion
	}
}

File: java/src/sdk/dataobjects/paypal/payflow/ECGetRequest.java
package paypal.payflow;

/**
 * Used for ExpressCheckout GET operation.
 * {@link ECSetRequest}
 * {@link ECDoRequest}
 */

public class ECGetRequest extends ExpressCheckoutRequest {

    /**
     * Constructor for ECGetRequest
     *
     * @param Token String
     *              <p/>
     *              <p/>
     *              ECGetRequest is used to set the data required for a Express Checkout GET operation.
     *              </p>
     *  .............
     * <p/>
     * Create the ECGetRequest object
     * ECGetRequest getEC = new ECGetRequest("[tokenid]");
     * <p/>
     * .............
     */
    public ECGetRequest(String Token) {
        super(PayflowConstants.PARAM_ACTION_GET, Token);
    }

    protected ECGetRequest(String Token, String Action) {
        super(PayflowConstants.PARAM_ACTION_GETBA, Token);
    }

}

